mbed_lstools/lstools_linux_generic.py

```diff
@@ -0,0 +1,311 @@
+                        result.append([mbed_name, mbed_dev_disk, mbed_mount_point, mbed_dev_serial, disk_hex_ids[dhi]])
+        return result
+
+    def get_not_detected(self, tids, disk_list, serial_list, mount_list):
+        """! Find all unknown mbed-enabled devices (may have 'mbed' string in USBID name)
+
+        @return list of lists [mbed_name, mbed_dev_disk, mbed_mount_point, mbed_dev_serial, disk_hex_id]
+
+        @param tids TargetID comprehensive list for detection (manufacturers_ids)
+        @param disk_list List of disks (mount points in /dev/disk)
+        @param serial_list List of serial devices (serial ports in /dev/serial)
+        @param mount_list List of lines from 'mount' command
+
+        @details Find for all disk connected all MBED ones we know about from TID list
+        """
+        map_tid_to_mbed = self.get_tid_mbed_name_remap(tids)
+        orphan_mbeds = []
+        for disk in disk_list:
+            if "mbed" in disk.lower():
+                orphan_found = True
+                for tid in map_tid_to_mbed.keys():
+                    if tid in disk:
+                        orphan_found = False
+                        break
+                if orphan_found:
+                    orphan_mbeds.append(disk)
+
+        # Search for corresponding MBED serial
+        disk_hex_ids = self.get_disk_hex_ids(orphan_mbeds)
+
+        result = []
+        # FInd orphan serial name
+        for dhi in disk_hex_ids:
+            orphan_serial = self.get_mbed_serial(serial_list, dhi)
+            if orphan_serial:
+                orphan_dev_disk = self.get_dev_name(disk_hex_ids[dhi])
+                orphan_dev_serial = '/dev/' + self.get_dev_name(orphan_serial)
+                orphan_mount_point = self.get_mount_point(orphan_dev_disk, mount_list)
+                if orphan_mount_point and orphan_dev_serial:
+                    result.append([None, orphan_dev_disk, orphan_mount_point, orphan_dev_serial, disk_hex_ids[dhi]])
+        return result
+
+    def get_tid_mbed_name_remap(self, tids):
+        """ Remap to get mapping:  ID -> mbed name
+        """
+        return tids
+
+    def get_dev_name(self, link):
+        """ Get device name from symbolic link list
+        """
+        device_sufix_pattern = ".*/([a-zA-Z0-9]*)$"
+        dsp = re.compile(device_sufix_pattern)
+        m = dsp.search(link)
+        mbed_dev = m.group(1) if m and len(m.groups()) else "unknown"
+        return mbed_dev
+
+    def get_mount_point(self, dev_name, mount_list):
+        """ Find mount points for MBED devices using mount command output
+
+        @return Returns None if mount point not found. Else returns device mount path
+
+        @param dev_name Device name (e.g 'sda')
+        @param mount_list List of all mounted devices (strings from Linux mount shell command)
+
+        @details
+        """
+        mount_media_pattern = "^/[a-zA-Z0-9/]*/" + dev_name  + " on (/[a-zA-Z0-9/]*) "
```

Comment:
there should be also `_` in end of detection pattern, like:
`"^/[a-zA-Z0-9/]*/" + dev_name  + " on (/[a-zA-Z0-9_/]*) "`.
This because looks that when multiple devices are connected mount point will be 
* `/media/MBED_` for second device
* `/media/MBED__` for third device
and so on...
* `/media/MBED______`  ... 
.